### PR TITLE
[FIX] point_of_sale: set missing fields in AML

### PR DESCRIPTION
**Pull request by nim-odoo**
- DB with Norwegian company
- Install account_saft
- Create a POS order
- Export General Ledger to SAF-T (Norway)

An error is raised because `price_total` and `price_subtotal` are
`None`.

AML created by the POS do not fill `exclude_from_invoice_tab`,
`price_total` and `price_subtotal`.

This commits aims at doing the bare minimum to fill the necessary fields
for the report.

Linked to odoo/enterprise#10906

opw-2263041

### CODE REVIEW

**Comment by nim-odoo** [comment]
@caburj @pimodoo This change could be considered quite big for a stable, knowing that it seems necessary only for a specific report in Norway.

What could be done is to move the call to `_get_price_total_and_subtotal_model` directly in `account_saft`. About `exclude_from_invoice_tab` it still makes sense to set it since it is set when creating an AML manually. 

cc @jco-odoo

**Comment by pimodoo** [comment]
@nim-odoo I don't really understand why _get_sale_vals are not managed like other lines. As it is not an invoice, it should only be "exclude_from_invoice_tab" No?

**Comment by caburj** [comment]
@nim-odoo I agree with @pimodoo 
The generated account move when closing a pos session is not technically an Invoice. Its sales line is a lump sum of the uninvoiced sales in the closed session. So maybe it is more appropriate to also exclude it from invoice tab.

**Comment by nim-odoo** [comment]
When you create an account move manually, `exclude_from_invoice_tab` is `False` while it is not an invoice. I set it to `False` to follow the same "logic".

File: addons/point_of_sale/models/pos_session.py [hunk 1]
```diff
@@ -668,10 +671,25 @@ def _get_sale_vals(self, key, amount, amount_converted):
             .mapped('invoice_repartition_line_ids' if sign == 1 else 'refund_repartition_line_ids')\
             .filtered(lambda line: line.repartition_type == 'base')\
             .tag_ids
+
+        # We get the prices total and subtotal since those can be used in reports
+        prices = self.env['account.move.line']._get_price_total_and_subtotal_model(
```
Comment (pimodoo) [inline addons/point_of_sale/models/pos_session.py:676]: Here we think it should be True too, and no need of the price stuff

**Comment by nim-odoo** [comment]
thanks for the feedback, I'll check directly on the report then

**Pull request closed without merging**